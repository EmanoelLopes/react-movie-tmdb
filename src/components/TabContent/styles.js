import styled from 'styled-components';
import { FadeIn } from 'utils/styled';

export const StyledTab = styled.section`
  position: absolute;
  visibility: hidden;

  &.tab-active {
    animation: ${FadeIn} 1s;
    position: relative;
    visibility: visible;
  }
`;
