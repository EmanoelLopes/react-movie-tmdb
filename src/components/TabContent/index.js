import React from 'react';
import { number, bool, node } from 'prop-types';
import { StyledTab } from './styles';

const TabContent = ({ index, isActive, content }) => {
  return (
    <StyledTab
      key={index}
      className={`tab-content ${(isActive) ? 'tab-active' : ''}`}>
      {content}
    </StyledTab>
  );
};

TabContent.propTypes = {
  index: number.isRequired,
  isActive: bool.isRequired,
  content: node.isRequired,
};

export default TabContent;
