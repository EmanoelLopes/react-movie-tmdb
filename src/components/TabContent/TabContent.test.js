import React from 'react';
import { mount } from 'enzyme';
import TabContent from 'components/TabContent';

let wrapped;
let defaultProps = {
 index: 0,
 isActive: true,
 content: '<div></div>',
};

beforeEach(() => {
  wrapped = mount(<TabContent {...defaultProps} />);
});

afterEach(() => {
  wrapped.unmount();
});

describe('<TabContent />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
