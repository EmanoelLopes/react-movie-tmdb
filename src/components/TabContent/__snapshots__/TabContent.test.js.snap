// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabContent /> should match snapshot 1`] = `
.c0 {
  position: absolute;
  visibility: hidden;
}

.c0.tab-active {
  -webkit-animation: eMLfYp 1s;
  animation: eMLfYp 1s;
  position: relative;
  visibility: visible;
}

<TabContent
  content="<div></div>"
  index={0}
  isActive={true}
>
  <styled.section
    className="tab-content tab-active"
    key="0"
  >
    <StyledComponent
      className="tab-content tab-active"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  position: absolute;
  visibility: hidden;

  &.tab-active {
    animation: ",
              Keyframes {
                "id": "sc-keyframes-eMLfYp",
                "inject": [Function],
                "name": "eMLfYp",
                "rules": Array [
                  "@-webkit-keyframes eMLfYp{from{opacity:0;}to{opacity:1;}}",
                  "@keyframes eMLfYp{from{opacity:0;}to{opacity:1;}}",
                ],
                "toString": [Function],
              },
              " 1s;
    position: relative;
    visibility: visible;
  }
",
            ],
          },
          "displayName": "styled.section",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "section",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <section
        className="tab-content tab-active c0"
      >
        &lt;div&gt;&lt;/div&gt;
      </section>
    </StyledComponent>
  </styled.section>
</TabContent>
`;
